system/HTTP/IncomingRequest.php

```diff
@@ -342,21 +342,23 @@ public function getJSON(bool $assoc = false, int $depth = 512, int $options = 0)
 	/**
 	 * Get a specific variable from a JSON input stream
 	 *
-	 * @param  string       $index  The variable that you want which can use dot syntax for getting specific values.
-	 * @param  boolean      $assoc  If true, return the result as an associative array.
-	 * @param  integer|null $filter Filter Constant
-	 * @param  mixed        $flags
+	 * @param  string             $index  The variable that you want which can use dot syntax for getting specific values.
+	 * @param  boolean            $assoc  If true, return the result as an associative array.
+	 * @param  integer|null       $filter Filter Constant
+	 * @param  array|integer|null $flags  Option
 	 * @return mixed
 	 */
-	public function getJsonVar(string $index, bool $assoc = false, $filter = null, $flags = null)
+	public function getJsonVar(string $index, bool $assoc = false, ?int $filter = null, $flags = null)
```

Comment:
```suggestion
	public function getJsonVar(string $index, bool $assoc = false, ?int $filter, $flags = null)
```

Reply:
1. A nullable type assumes NULL, but does not define a default.
2. Violating php 8 compatibility